invenio_cli/helpers/commands.py

```diff
@@ -0,0 +1,135 @@
+        :param local: boolean True if local environment
+        """
+        if local:
+            self.environment = LocalCommands(cli_config)
+        else:
+            self.environment = ContainerizedCommands()
+
+    def build(self, pre, lock):
+        """Delegates to build of environment commands."""
+        return self.environment.build(pre, lock)
+
+
+class LocalCommands(object):
+    """Local environment CLI commands."""
+
+    def __init__(self, cli_config):
+        """Constructor."""
+        self.cli_config = cli_config
+
+    def _install_py_dependencies(self, pre, lock):
+        """Install Python dependencies."""
+        click.secho('Installing python dependencies...', fg='green')
+        command = ['pipenv', 'install', '--dev']
+        if pre:
+            command += ['--pre']
+        if not lock:
+            command += ['--skip-lock']
+        subprocess.run(command)
+
+    def _update_instance_path(self):
+        """Update path to instance in config."""
+        path = subprocess.run(
+            ['pipenv', 'run', 'invenio', 'shell', '--no-term-title',
+                '-c', '"print(app.instance_path, end=\'\')"'],
+            check=True, universal_newlines=True, stdout=subprocess.PIPE
+        ).stdout.strip()
+        self.cli_config.update_instance_path(path)
+
+    def _symlink_project_config(self):
+        project_config = 'invenio.cfg'
+        click.secho("Symlinking {}...".format(project_config), fg="green")
+        target_path = self.cli_config.get_project_dir() / project_config
+        link_path = self.cli_config.get_instance_path() / project_config
+        filesystem.force_symlink(target_path, link_path)
+
+    def _symlink_templates(self):
+        """Symlink the templates folder."""
+        templates = 'templates'
+        click.secho('Symlinking {}/...'.format(templates), fg='green')
+        target_path = self.cli_config.get_project_dir() / templates
+        link_path = self.cli_config.get_instance_path() / templates
+        filesystem.force_symlink(target_path, link_path)
+
+    def _copy_statics_and_assets(self):
+        """Copy project's statics and assets into instance dir."""
+        click.secho('Copying project statics and assets...', fg='green')
+
+        static = 'static'
+        src_dir = self.cli_config.get_project_dir() / static
+        src_dir = str(src_dir)  # copy_tree below doesn't accept Path objects
+        dst_dir = self.cli_config.get_instance_path() / static
+        dst_dir = str(dst_dir)
+        # using it for a different purpose then intended but very useful
+        dir_util.copy_tree(src_dir, dst_dir)
+
+        # TODO: Uncomment when rebased on @zzachero work
+        # assets = 'assets'
+        # src_dir = self.cli_config.get_project_dir() / assets
+        # src_dir = str(src_dir)
+        # dst_dir = self.cli_config.get_instance_path() / assets
+        # dst_dir = str(dst_dir)
+        # dir_util.copy_tree(src_dir, dst_dir)
```

Comment:
Just thinking that if people call  the `build` command again while everything is already built then the `copy_tree` command will fail as it requires `dst` folder not to exist. Also I think that the instance assets should be linked or we need to manually copy over in every change. This could fine for now but I am just saying that in some point we should integrate this behaviour :)

Reply:
> Just thinking that if people call the build command again while everything is already built then the copy_tree command will fail as it requires dst folder not to exist.

Ah yes I faced that issue with `shutil.copytree` too. That's why I went with `distutils.dir_util.copy_tree` here. It doesn't have that problem.

+1 for the symlinking approach in the future. We are putting it on the backburner for now.